### PR TITLE
Suggestion creation

### PR DESCRIPTION
**Pull request by AgeX21**
This PR fills the requirement for the issue associated : 
- A composable to enter information to create a suggestion and send it to the database
- Certain information can be prefilled when creating the screen to allow other screens to have more control on this composable

### CODE REVIEW

File: app/src/androidTest/java/com/github/se/wanderpals/screens/CreateSuggestionScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.compose.ui.test.SemanticsNodeInteractionsProvider
+import io.github.kakaocup.compose.node.element.ComposeScreen
+import io.github.kakaocup.compose.node.element.KNode
+
+class CreateSuggestionScreenTest(semanticsProvider: SemanticsNodeInteractionsProvider) :
+    ComposeScreen<CreateSuggestionScreenTest>(
+        semanticsProvider = semanticsProvider,
+        viewBuilderAction = { hasTestTag("createSuggestionScreen") }) {
+
+  val screenTitle: KNode = onNode { hasTestTag("createSuggestionTitle") }
```
Comment (annhit29) [inline app/src/androidTest/java/com/github/se/wanderpals/screens/CreateSuggestionScreenTest.kt:12]: You can create a test to use the `screenTitle` in order to get a higher coverage on SonarCloud

and a detail: change the class name from `CreateSuggestionScreenTest.kt` to `CreateSuggestionScreen.kt`

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/CreateSuggestionViewModel.kt
Comment (annhit29) [inline app/src/main/java/com/github/se/wanderpals/model/viewmodel/CreateSuggestionViewModel.kt:1]: to be checked with the database

**Review by annhit29 — CHANGES_REQUESTED** [review]
The functional and the visual parts are good. 
Please read the comments in the CreateSuggestionScreenTest.kt and CreateSuggestionViewModel.kt files.

The test coverage is now at 53.7%. You can try to add more tests!

All the tests passed on my side :D

File: app/src/androidTest/java/com/github/se/wanderpals/screens/CreateSuggestionScreenTest.kt (hunk 1 above)
Comment (AgeX21) [reply app/src/androidTest/java/com/github/se/wanderpals/screens/CreateSuggestionScreenTest.kt]: Thanks for the comment,
I changed the class name like suggested.

File: app/src/androidTest/java/com/github/se/wanderpals/screens/CreateSuggestionScreen.kt
Comment (annhit29) [inline app/src/androidTest/java/com/github/se/wanderpals/screens/CreateSuggestionScreen.kt:1]: Good for consistency :D

**Review by annhit29 — APPROVED** [review]
All the tests passed on my end!

54.2% of coverage can be improved. 
You can test MyTimePickerDialog and TimePickerDialog composables of the class CreateSuggestion.kt later.
For CreateSuggestionViewModel.kt, the reason why the code in this file is uncovered is that you use onSuccess() to see if the operation is working correctly.

You can remove the part of the code you have commented out if you don't (plan to) use it in the future.

The list of parameters of the composable CreateSuggestion can be reduced. You can keep it for now and reduce it when you move on the project and see which parameters of the list can be removed.

**Pull request merged**